main/src/ui/chat_input/chat_input_controller.vala

```diff
@@ -48,12 +48,32 @@ public class ChatInputController : Object {
         chat_text_view_controller.initialize_for_conversation(conversation);
         
         Xmpp.Jid? own_jid = stream_interactor.get_module(MucManager.IDENTITY).get_own_jid(conversation.counterpart, conversation.account);
+        
         if (conversation.type_ == conversation.Type.GROUPCHAT){
             if (stream_interactor.get_module(MucManager.IDENTITY).is_moderated_room(conversation.account, conversation.counterpart) && 
             stream_interactor.get_module(MucManager.IDENTITY).get_role(own_jid, conversation.account)==Xmpp.Xep.Muc.Role.VISITOR) {
-                set_input_field_status(new Plugins.InputFieldStatus("This is a moderated group, you don't have voice", Plugins.InputFieldStatus.MessageType.ERROR, Plugins.InputFieldStatus.InputState.NO_SEND));
+                set_input_field_status(new Plugins.InputFieldStatus("This conference does not allow you to send messages. <a href=\"nonexistant\"> Request permission </a>", Plugins.InputFieldStatus.MessageType.ERROR, Plugins.InputFieldStatus.InputState.NO_SEND, true));
             }
         }
+
+        stream_interactor.get_module(MucManager.IDENTITY).received_occupant_role.connect((account, jid, role) => {
+            if (conversation.type_ == conversation.Type.GROUPCHAT){
+                if (stream_interactor.get_module(MucManager.IDENTITY).is_moderated_room(conversation.account, conversation.counterpart) && 
+                stream_interactor.get_module(MucManager.IDENTITY).get_role(own_jid, conversation.account)!=Xmpp.Xep.Muc.Role.VISITOR && jid.equals(own_jid)) {
+                    reset_input_field_status();
+                }
+        }
+        });
+
+        stream_interactor.get_module(MucManager.IDENTITY).room_info_updated.connect((account, muc_jid) => {
+            if (conversation.type_ == conversation.Type.GROUPCHAT){
+                if (stream_interactor.get_module(MucManager.IDENTITY).is_moderated_room(conversation.account, conversation.counterpart) && 
+                stream_interactor.get_module(MucManager.IDENTITY).get_role(own_jid, conversation.account)==Xmpp.Xep.Muc.Role.VISITOR) {
+                    set_input_field_status(new Plugins.InputFieldStatus("This conference does not allow you to send messages. <a href=\"nonexistant\"> Request permission </a>", Plugins.InputFieldStatus.MessageType.ERROR, Plugins.InputFieldStatus.InputState.NO_SEND, true));
+                }
+                else reset_input_field_status();
```

Comment:
Code style: Please don't mix braceless else with braced if (or other way round). Use braces in this case:
```vala
} else {
  reset_input_field_status();
}
```